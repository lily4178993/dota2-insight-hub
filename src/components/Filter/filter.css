.header-filter-box {
  position: relative;
  display: flex;
  justify-content: space-between;
  padding: 0.5em 2em;
  text-transform: uppercase;
  font-family: var(--font-family-2);
  background: var(--color-blue-4);
}

.header-filter-box button {
  font-family: var(--font-family-2);
  font-size: 0.8em;
}

.header-filter-box button i {
  vertical-align: bottom;
  font-size: 1rem;
  transition: transform 0.9s ease;
}

.filter-box {
  position: absolute;
  bottom: 100%;
  right: 1em;
  padding: 0.2em 1.5em;
  text-align: right;
  background: var(--color-blue-4);
  z-index: 1;
  transition: all 500ms ease-in-out;
}
