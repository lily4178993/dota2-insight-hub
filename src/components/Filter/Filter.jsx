import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './filter.css';

const Filter = ({ setFilter }) => {
  const [isBoxFilterOpen, setIsBoxFilterOpen] = useState(true);

  return (
    <div className="header-filter-box">
      <span>Choose a Card</span>
      <button type="button" title="Filter" onClick={() => setIsBoxFilterOpen(!isBoxFilterOpen)}>
        {isBoxFilterOpen ? 'Filter' : 'Filter By'}
        {' '}
        <i
          className="bx bx-down-arrow-circle"
          style={{ transform: isBoxFilterOpen ? 'rotate(0deg)' : 'rotate(180deg)' }}
        />
      </button>
      <div
        className="filter-box"
        style={{
          right: isBoxFilterOpen ? '-100%' : '1em',
        }}
      >
        <ul>
          <li>
            <button
              type="button"
              title="Filter by Names"
              onClick={() => { setFilter('Names'); setIsBoxFilterOpen(!isBoxFilterOpen); }}
            >
              Names
            </button>
          </li>
          <li>
            <button
              type="button"
              title="Filter by Counts"
              onClick={() => { setFilter('Counts'); setIsBoxFilterOpen(!isBoxFilterOpen); }}
            >
              Counts
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
};

Filter.propTypes = {
  setFilter: PropTypes.func,
};

Filter.defaultProps = {
  setFilter: PropTypes.func,
};

export default Filter;
