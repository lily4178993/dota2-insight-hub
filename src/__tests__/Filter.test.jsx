import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Filter from '../components/Filter/Filter';

test('Filter component matches snapshot', () => {
  const setFilter = jest.fn();
  const { container, getByTitle } = render(<Filter setFilter={setFilter} />);

  const filterButton = getByTitle('Filter');
  fireEvent.click(filterButton);

  const filterByNameButton = getByTitle('Filter by Names');
  const filterByCountsButton = getByTitle('Filter by Counts');

  expect(container).toMatchSnapshot();

  fireEvent.click(filterByNameButton);
  expect(container).toMatchSnapshot();

  fireEvent.click(filterButton);
  fireEvent.click(filterByCountsButton);
  expect(container).toMatchSnapshot();
});
