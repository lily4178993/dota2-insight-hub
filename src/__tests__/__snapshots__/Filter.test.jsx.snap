// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component matches snapshot 1`] = `
<div>
  <div
    class="header-filter-box"
  >
    <span>
      Choose a Card
    </span>
    <button
      title="Filter"
      type="button"
    >
      Filter By
       
      <i
        class="bx bx-down-arrow-circle"
        style="transform: rotate(180deg);"
      />
    </button>
    <div
      class="filter-box"
      style="right: 1em;"
    >
      <ul>
        <li>
          <button
            title="Filter by Names"
            type="button"
          >
            Names
          </button>
        </li>
        <li>
          <button
            title="Filter by Counts"
            type="button"
          >
            Counts
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Filter component matches snapshot 2`] = `
<div>
  <div
    class="header-filter-box"
  >
    <span>
      Choose a Card
    </span>
    <button
      title="Filter"
      type="button"
    >
      Filter
       
      <i
        class="bx bx-down-arrow-circle"
        style="transform: rotate(0deg);"
      />
    </button>
    <div
      class="filter-box"
      style="right: -100%;"
    >
      <ul>
        <li>
          <button
            title="Filter by Names"
            type="button"
          >
            Names
          </button>
        </li>
        <li>
          <button
            title="Filter by Counts"
            type="button"
          >
            Counts
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Filter component matches snapshot 3`] = `
<div>
  <div
    class="header-filter-box"
  >
    <span>
      Choose a Card
    </span>
    <button
      title="Filter"
      type="button"
    >
      Filter
       
      <i
        class="bx bx-down-arrow-circle"
        style="transform: rotate(0deg);"
      />
    </button>
    <div
      class="filter-box"
      style="right: -100%;"
    >
      <ul>
        <li>
          <button
            title="Filter by Names"
            type="button"
          >
            Names
          </button>
        </li>
        <li>
          <button
            title="Filter by Counts"
            type="button"
          >
            Counts
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
